secretstores/azure/keyvault/keyvault.go

```diff
@@ -45,6 +45,29 @@ func NewAzureKeyvaultSecretStore(logger logger.Logger) secretstores.SecretStore
 
 // Init creates a Azure Key Vault client
 func (k *keyvaultSecretStore) Init(metadata secretstores.Metadata) error {
+	// Fix for maintaining backwards compatibility with a change introduced in 1.3 that allowed specifying an Azure environment by setting a FQDN for vault name
+	// This should be considered deprecated and users should rely the "azureEnvironment" metadata instead, but it's maintained here for backwards-compatibility
+	if vaultName, ok := metadata.Properties[componentVaultName]; ok {
+		keyVaultSuffixToEnvironment := map[string]string{
+			".vault.azure.net":         "AZUREPUBLICCLOUD",
```

Comment:
Note that the list of suffixes is hardcoded here because it was in 1.3. This will support any (and all) the ones that were supported by 1.3